lib/rucio/daemons/replicarecoverer/suspicious_replica_recoverer.py

```diff
@@ -80,13 +73,32 @@ def declare_suspicious_replicas_bad(once: bool = False, younger_than: int = 3, n
     :returns: None
     """
 
+    run_daemon(
+        once=once,
+        graceful_stop=GRACEFUL_STOP,
+        executable=argv[0],
+        logger_prefix='replica_recoverer: ',
+        partition_wait_time=1,
+        sleep_time=sleep_time,
+        run_once_fnc=functools.partial(
+            run_once,
+            younger_than=younger_than,
+            nattempts=nattempts,
+            vos=vos,
+            limit_suspicious_files_on_rse=limit_suspicious_files_on_rse,
+            active_mode=active_mode
+        ),
+    )
+
+
+def run_once(heartbeat_handler: Any, younger_than: int, nattempts: int, vos: List[str], limit_suspicious_files_on_rse: int, active_mode: int, **_kwargs):
     # assembling the worker name identifier ('executable') including the rses from <rse_expression>
     # in order to have the possibility to detect a start of a second instance with the same set of RSES
 
-    executable = argv[0]
+    worker_number, total_workers, logger = heartbeat_handler.live()
 
-    prepend_str = 'replica_recoverer: '
-    logger = formatted_logger(logging.log, prepend_str + '%s')
+    # wait a moment in case all workers started at the same time
+    GRACEFUL_STOP.wait(1)
```

Comment:
I have the impression that this is unnecessary and should be left to `partition_wait_time` instead, see https://github.com/rucio/rucio/blob/771a80183a60aff917031d8b182f574d35f325e1/lib/rucio/daemons/common.py#L112